include/netutils/webclient.h

```diff
@@ -97,6 +97,140 @@
 typedef void (*wget_callback_t)(FAR char **buffer, int offset,
                                 int datend, FAR int *buflen, FAR void *arg);
 
+/* webclient_sink_callback_t: callback to consume data
+ *
+ * Same as wget_callback_t, but allowed to fail.
+ *
+ * Input Parameters:
+ *   Same as wget_callback_t
+ *
+ * Return value:
+ *   0 on success.
+ *   A negative errno on error.
+ */
+
+typedef CODE int (*webclient_sink_callback_t)(FAR char **buffer, int offset,
+                                              int datend, FAR int *buflen,
+                                              FAR void *arg);
+
+/* webclient_body_callback_t: a callback to provide request body
+ *
+ * This callback can be called multiple times to provide
+ * webclient_context::bodylen bytes of the data.
+ * It's the responsibility of this callback to maintain the current
+ * "offset" of the data. (similarly to fread(3))
+ *
+ * An implementation of this callback should perform either of
+ * the followings:
+ *
+ * - fill the buffer (specified by buffer and *sizep) with the data
+ * - update *datap with a buffer filled with the data
+ *
+ * Either ways, it should update *sizep to the size of the data.
+ *
+ * A short result is allowed. In that case, this callback will be called
+ * again to provide the remaining data.
+ *
+ * Input Parameters:
+ *   buffer - The buffer to fill.
+ *   sizep  - The size of buffer/data in bytes.
+ *   datap  - The data to return.
```

Comment:
to allow users to pass their own buffers directly.
the existing wget_callback_t prototype made me think it can be important for some users to avoid extra buffers.
i myself don't need it.